### PR TITLE
retry getting the shared build location

### PR DESCRIPTION
**Pull request by BenTheElder**
The canary job [flaked once](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/51707/pull-kubernetes-e2e-gce-canary/14/) because we don't yet retry getting the shared build location, the file it tried to fetch exists so hopefully simply retrying a few times will help.

### CODE REVIEW

File: scenarios/kubernetes_e2e.py [hunk 1]
```diff
@@ -439,6 +440,13 @@ def read_gcs_path(gcs_path):
     print >>sys.stderr, "Read GCS Path: %s" % loc
     return loc
 
+def get_gcs_path(args):
```
Comment (krzyzacy) [inline scenarios/kubernetes_e2e.py]: you only need `args.use_shared_build` and `args.gcs_shared` here, don't have to pass in all the args.

and please add a function docstring

File: scenarios/kubernetes_e2e.py [hunk 2]
```diff
@@ -485,18 +493,21 @@ def main(args):
...
+        gcs_path = get_gcs_path(args)
         print >>sys.stderr, 'Getting shared build location from: '+gcs_path
-        try:
-            # tell kubetest to extract from this location
-            args.kubetest_args.append('--extract=' + read_gcs_path(gcs_path))
-            args.build = None
-        except urllib2.URLError as err:
-            raise RuntimeError('Failed to get shared build location: %s' % err.reason)
+        retries = 0
+        while True:
```
Comment (krzyzacy) [inline scenarios/kubernetes_e2e.py]: maybe for/else will be more clear here

Comment (BenTheElder) [reply scenarios/kubernetes_e2e.py]: hmm ok `for _ in xrange(10):` could work, but I'm not sure if that's clearer.

File: scenarios/kubernetes_e2e.py (hunk 1 above)
Comment (BenTheElder) [reply scenarios/kubernetes_e2e.py]: agreed, the method above needs a docstring too.

File: scenarios/kubernetes_e2e.py (hunk 2 above)
Comment (BenTheElder) [reply scenarios/kubernetes_e2e.py]: I think the main problem is it makes using the actual error ugly. I want to ignore the error the first n times and then give up and give the error.

**Pull request merged**